hack/generate-metrics-doc.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2019 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/projectcontour/contour/internal/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// Collect all the label names for this metric and return them as
+// a comma-deparated string.
+func labels(mf *dto.MetricFamily) string {
+	var l []string
+
+	for _, m := range mf.GetMetric() {
+		for _, pair := range m.GetLabel() {
+			l = append(l, pair.GetName())
+		}
+	}
+
+	return strings.Join(l, ", ")
+}
+
+// Generate a string name for the metric type, linking to the
+// Prometheus docs if we know there is a suitable target.
+func typeof(mf *dto.MetricFamily) string {
+	switch t := mf.GetType(); t {
+	case dto.MetricType_COUNTER:
```

Comment:
wil it work if you stack the case statements here?
```
case dto.MetricType_COUNTER, 
         dto.MetricType_SUMMARY,
         dto.MetricType_GAUGE,
         dto.MetricType_HISTOGRAM:
```

Reply:
Fixed.